internal/metrics/metrics.go

```diff
@@ -127,3 +129,44 @@ func (s *ScheduleTimer) Mark(ctx context.Context, role, outcome string) {
 	metrics.FromContext(ctx).Histogram(
 		schedulingRetryTime, sparkRoleTag, outcomeTag, s.instanceGroupTag, s.retryTag).Update(now.Sub(s.lastSeenTime).Nanoseconds())
 }
+
+// ReportCrossZoneMetric reports metric about cross AZ traffic between pods of a spark application
+func ReportCrossZoneMetric(ctx context.Context, driverNodeName string, executorNodeNames []string, nodes []*v1.Node) {
+	executorNodesSet := make(map[string]bool)
+	for _, n := range executorNodeNames {
+		executorNodesSet[n] = true
+	}
+
+	zonesCounter := make(map[string]int)
+	for _, n := range nodes {
+		if _, ok := executorNodesSet[n.Name]; ok {
+			executorZone, ok := n.Labels[nodeZoneLabel]
+			if !ok {
+				svc1log.FromContext(ctx).Warn("zone label not found for node", svc1log.SafeParam("nodeName", n.Name))
+				return
```

Comment:
why do we return here. does a single node not having the label mean we don't output this metric at all? that seems bad?